Sources/Fullscreen/Consent/Details/ConsentDetailView.swift

```diff
@@ -0,0 +1,115 @@
+//
+//  Copyright © 2018 FINN AS. All rights reserved.
+//
+import UIKit
+
+public protocol ConsentDetailViewDelegate: class {
+    func consentDetailView(_ consentDetailView: ConsentDetailView, didToggleSwitch position: Bool)
+    func readMoreButtonPressed(in consentDetailView: ConsentDetailView)
+}
+
+public struct ConsentDetailViewModel {
+    public let heading: String
+    public let definition: String
+    public let indexPath: IndexPath
+
+    public init(heading: String, definition: String, indexPath: IndexPath) {
+        self.heading = heading
+        self.definition = definition
+        self.indexPath = indexPath
+    }
+}
+
+public class ConsentDetailView: UIView {
+
+    public var model: ConsentDetailViewModel? {
+        didSet { set(model: model) }
+    }
+
+    public var buttonTitle: String? {
+        didSet { button.setTitle(buttonTitle, for: .normal) }
+    }
+
+    public weak var delegate: ConsentDetailViewDelegate?
+    public var state: Bool = false {
+        didSet { theSwitch.setOn(state, animated: false) }
+    }
+
+    lazy var textView: Label = {
```

Comment:
Could this label and components below be private?

Reply:
They can and they will be